### DISCUSSION TITLE
Progress at 'Goal' level

### DISCUSSION THREAD

**Discussion by phillipgardiner** [open-sdg/open-sdg#1354] [Ideas]
Recently a piece of worked has been kicked off to provide countries with a minimum viable product (MVP) way of reporting 'progress'. Currently this work covers reporting progress at an indicator level (https://github.com/open-sdg/open-sdg/issues/1321).

It has been identified that additionally there is a user need to be able to provide progress at the goals level (e.g. _As a concerned citizen, I want a high-level view of how the UK is doing against the SDGs, so that I can understand if we're achieving the goals_).

This discussion thread has been kicked off in order to try the new 'discussion' functionality but also provide background to the work already completed in this area.

In December 2020 a design review study was undertaken where participants were probed (one-on-one) to explain their understanding of mock-ups (sketches) shown. This included a mock up of how progress could be shown at a goal / summary level. Following this study and using the feedback received, the mock ups shown to participants were iterated and re-designed.

This iterated design along with supporting notes can be viewed here for discussion: 
[Goal Progress - Radar Chart.pdf](https://github.com/open-sdg/open-sdg/files/7236373/Goal.Progress.-.Radar.Chart.pdf)

**Comment by brockfanning** [comment]
Just an update, as of version 1.6.0 (just released) pages 2 and 3 of the PDF linked above are implemented - I believe. So, progress at the indicator level is now part of the platform. But as @phillipgardiner mentions above the next thing to discuss is progress at the *goal* level.

Personally I like the "radar chart" approach on page 1 of the PDF. It seems like an elegant approach. I feel like mobile would work fine, since it doesn't need to be super big to get the idea across. We may have to get creative with how we would communicate the progress to screen readers.

Just to play devil's advocate and consider alternatives - an approach that might be closer to existing designs is something like the reporting status page. Only, instead of the horizontal bar showing a breakdown of status types, it could be a horizontal percentage bar from 0 to 100 for that goal.

But again I also like that radar design. I know that radar charts are [possible in Chart.js](https://www.chartjs.org/docs/latest/samples/other-charts/radar.html) so it is technically feasible.

**Reply by phillipgardiner** [reply]
Hey, @brockfanning just as a bit more background.. As part of the research we've undertaken previously we did evaluate something that relates to your idea about horizontal bars, alongside the radial chart idea (see example below). We found that this did not evaluate as well with end users, primarily as they found it harder to interpret. Where as the radial idea was seen as simple and great at providing a quick overview - obviously these comparisons were done without any thought around technical constraints and accessibility etc as it was more about concepts.

![image](https://user-images.githubusercontent.com/60919820/142643229-7e3d7bd1-c01b-4645-ba60-8b0508f88615.png)

**Comment by brockfanning** [comment]
@phillipgardiner Thanks for that background. One followup about that bar-chart approach: What were the up/down/right arrows meant to show? I'm just asking in case that's something that also needs to be on the radar approach.

Aside from the look-and-feel question, I'm also thinking about the data side of things. Specifically, how does the platform know which percentage to give each goal?

Right now we have basic per-indicator functionality, where each indicator is given an arbitrary "progress status". The default status types are "not available", "challenges remain", "approaching target", and "target achieved".

My first thought is some kind of point system, like:

* not available: 0 points
* challenges remain: 1 point
* approach target: 2 points
* target achieved: 3 points

So the percentage which displays for a goal on the radar chart would be a proportion of the total points for the actual indicators in the goal, out of the total points if all were "target achieved".

For example, if there are 100 indicators in a goal (for easier math...) out of which 25 are "not available" (0 points), 25 are "challenges remain" (25 points), 25 are "approaching target" (50 points), and 25 are "target achieved" (75 points), then that goal would have 150 points, of a possible 300 points; and so their percentage would be 50%.
